plugins/meta/iptables-allow/iptables_allow.go

```diff
@@ -0,0 +1,190 @@
+	if conf.RawPrevResult != nil {
+		resultBytes, err := json.Marshal(conf.RawPrevResult)
+		if err != nil {
+			return nil, fmt.Errorf("could not serialize prevResult: %v", err)
+		}
+		res, err := version.NewResult(conf.CNIVersion, resultBytes)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse prevResult: %v", err)
+		}
+		conf.RawPrevResult = nil
+		conf.PrevResult, err = current.NewResultFromResult(res)
+		if err != nil {
+			return nil, fmt.Errorf("could not convert result to current version: %v", err)
+		}
+	}
+
+	return &conf, nil
+}
+
+func getPrivChainRules(intf string) [][]string {
+	var rules [][]string
+	rules = append(rules, []string{"-i", intf, "!", "-o", intf, "-j", "ACCEPT"})
+	rules = append(rules, []string{"-i", intf, "-o", intf, "-j", "ACCEPT"})
+	rules = append(rules, []string{"-o", intf, "-m", "conntrack", "--ctstate", "RELATED,ESTABLISHED", "-j", "ACCEPT"})
+	return rules
+}
+
+func ensureChain(ipt *iptables.IPTables, table, chain string) error {
+	chains, err := ipt.ListChains(table)
+	if err != nil {
+		return fmt.Errorf("failed to list iptables chains: %v", err)
+	}
+	for _, ch := range chains {
+		if ch == chain {
+			return nil
+		}
+	}
+
+	return ipt.NewChain(table, chain)
+}
+
+func addRules(intf string, proto iptables.Protocol) error {
+	var (
+		err    error
+		ipt    *iptables.IPTables
+		exists bool
+	)
+
+	ipt, err = iptables.NewWithProtocol(proto)
+	if err != nil {
+		return fmt.Errorf("failed to initialize iptables helper: %v", err)
+	}
+
+	adminChainName := fmt.Sprintf("CNI-ADMIN-%s", intf)
+	privChainName := fmt.Sprintf("CNI-FORWARD-%s", intf)
+
+	var filterRules [][]string
+	filterRules = append(filterRules, []string{"-m", "comment", "--comment", fmt.Sprintf("CNI interface %s private rules", intf), "-j", privChainName})
+	filterRules = append(filterRules, []string{"-i", intf, "!", "-o", intf, "-m", "comment", "--comment", fmt.Sprintf("CNI interface %s administrator overrides", intf), "-j", adminChainName})
+
+	defer func() {
+		// Clean up on error
+		if err != nil {
+			ipt.DeleteChain("filter", privChainName)
+			ipt.DeleteChain("filter", adminChainName)
+			for _, rule := range filterRules {
+				ipt.Delete("filter", "FORWARD", rule...)
+			}
+		}
+	}()
+
+	if err := ensureChain(ipt, "filter", privChainName); err != nil {
+		return err
+	}
+	if err := ensureChain(ipt, "filter", adminChainName); err != nil {
+		return err
+	}
+
+	for _, rule := range filterRules {
+		exists, err = ipt.Exists("filter", "FORWARD", rule...)
+		if !exists && err == nil {
+			err = ipt.Insert("filter", "FORWARD", 1, rule...)
+		}
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, rule := range getPrivChainRules(intf) {
+		if err := ipt.AppendUnique("filter", privChainName, rule...); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func cmdAdd(args *skel.CmdArgs) error {
+	iptConf, err := parseConf(args.StdinData)
+	if err != nil {
+		return err
+	}
+
+	var intfName string
+	if iptConf.PrevResult != nil {
```

Comment:
This looks like it's relying on ordering to pick the "bridge" interface. Does it make sense to also accept the ifname as a configuration hint?

Reply:
@squeed yes, I'll do this as an option with fallback to the first PrevResult host interface.  This will also make the plugin compatible with 0.1.0 and later too.